tools/osbuild-depsolve-dnf5

```diff
@@ -0,0 +1,500 @@
+
+    return urls[0]
+
+
+class TransactionError(Exception):
+    pass
+
+
+class RepoError(Exception):
+    pass
+
+
+class Solver():
+    """Solver implements package related actions
+
+    These include depsolving a package set, searching for packages, and dumping a list
+    of all available packages.
+    """
+
+    # pylint: disable=too-many-arguments
+    def __init__(self, repos, module_platform_id, persistdir, cachedir, arch, exclude_pkgs=None):
+        if not exclude_pkgs:
+            exclude_pkgs = []
+
+        # substitutions to use with the repo urls
+        self.substitutions = {
+            "arch": arch,
+            "basearch": _BASEARCH_MAP[arch]
+        }
+        self.base = dnf5.base.Base()
+        self.base.get_vars().set("arch", arch)
+
+        # Enable fastestmirror to ensure we choose the fastest mirrors for
+        # downloading metadata (when depsolving) and downloading packages.
+        conf = self.base.get_config()
+        conf.fastestmirror = True
+
+        # Weak dependencies are installed for the 1st transaction
+        # This is set to False for any subsequent ones in depsolve()
+        conf.install_weak_deps = True
+
+        # We use the same cachedir for multiple architectures. Unfortunately,
+        # this is something that doesn't work well in certain situations
+        # with zchunk:
+        # Imagine that we already have cache for arch1. Then, we use dnf-json
+        # to depsolve for arch2. If ZChunk is enabled and available (that's
+        # the case for Fedora), dnf will try to download only differences
+        # between arch1 and arch2 metadata. But, as these are completely
+        # different, dnf must basically redownload everything.
+        # For downloding deltas, zchunk uses HTTP range requests. Unfortunately,
+        # if the mirror doesn't support multi range requests, then zchunk will
+        # download one small segment per a request. Because we need to update
+        # the whole metadata (10s of MB), this can be extremely slow in some cases.
+        # I think that we can come up with a better fix but let's just disable
+        # zchunk for now. As we are already downloading a lot of data when
+        # building images, I don't care if we download even more.
+        conf.zchunk = False
+
+        # Set the rest of the dnf configuration.
+        conf.module_platform_id = module_platform_id
+        conf.config_file_path = "/dev/null"
+        conf.persistdir = persistdir
+        conf.cachedir = cachedir
+
+        # Load the file lists, so dependency on paths will work
+        conf.optional_metadata_types = ['comps', 'filelists']
+
+        # Cannot modify .conf() values after this
+        self.base.setup()
+
+        try:
+            # NOTE: With libdnf5 packages are excluded in the repo setup
+            for repo in repos:
+                self._dnfrepo(repo, exclude_pkgs)
+
+            self.base.get_repo_sack().update_and_load_enabled_repos(False)
```

Comment:
Would it be possible to use a keyword args for this `False` to make it clearer what the bool is about? Probably `self.base.get_repo_sack().update_and_load_enabled_repos(load_system=False)`

Reply:
Oh hey, that still works. Will do.